sqlmesh/utils/dag.py

```diff
@@ -99,7 +99,9 @@ def sorted(self) -> t.List[T]:
                 for deps in unprocessed_nodes.values():
                     deps -= next_nodes
 
-                self._sorted.extend(next_nodes)
+                # Sort to make the order deterministic
+                # TODO: Make protocol that makes the type var both hashable and sortable once we are on Python 3.8+
+                self._sorted.extend(sorted(next_nodes))  # type: ignore
```

Comment:
why was this necessary?

Reply:
Well I had a bug in my logic at one point and it would only show up sometimes and I found that this was a source for that inconsistency. So it is necessary in the sense that it makes the behavior of the system more consistent.